crates/notion-core/src/distro/mod.rs

```diff
@@ -33,9 +34,35 @@ impl<V> Fetched<V> {
     }
 }
 
-pub trait Distro: Sized {
-    type VersionDetails;
+/// Abstraction to contain info about Distro versions.
+#[derive(Eq, PartialEq, Clone, Debug)]
+pub enum DistroVersion {
+    // the version of the Node runtime, and the npm version installed with that
+    Node(Version, Version),
+    Yarn(Version),
+    Npm(Version),
+    Npx(Version),
```

Comment:
I wonder if we should not have the `Npx` variant, since if we want to support installing it independently from npm, it would just be as a package, so you could use the `Package` variant for that.

Reply:
removed the Npx variants - those can use the `Package` variant